src/storage/mvcc/reader/backward_scanner.rs

```diff
@@ -0,0 +1,713 @@
+        let last_handled_key = self.write_cursor.key(&mut self.statistics.write).to_vec();
+
+        let seek_key = user_key.clone().append_ts(safe_ts);
+        assert!(seek_key.encoded().as_slice() < last_handled_key.as_slice());
+
+        // TODO: Replace by cast + seek().
+        self.write_cursor
+            .internal_seek(&seek_key, &mut self.statistics.write)?;
+        assert!(self.write_cursor.valid());
+
+        loop {
+            // After seek, or after some `next()`, we may reach `last_handled_key` again. It
+            // means we have checked all versions for this user key. We use `last_version` as
+            // return.
+            let mut is_done = false;
+            {
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                // We should never reach another user key.
+                assert!(Key::is_user_key_eq(
+                    current_key,
+                    user_key.encoded().as_slice()
+                ));
+                if current_key >= last_handled_key.as_slice() {
+                    // We reach the last handled key,
+                    is_done = true;
+                }
+            }
+            if is_done {
+                return Ok(self.handle_last_version(last_version, user_key)?);
+            }
+
+            let write = Write::parse(self.write_cursor.value(&mut self.statistics.write))?;
+            self.statistics.write.processed += 1;
+
+            match write.write_type {
+                WriteType::Put => {
+                    return Ok(Some(self.reverse_load_data_by_write(write, user_key)?))
+                }
+                WriteType::Delete => return Ok(None),
+                WriteType::Lock | WriteType::Rollback => {
+                    // Continue iterate next `write`.
+                    self.write_cursor.next(&mut self.statistics.write);
+                    assert!(self.write_cursor.valid());
+                }
+            }
+        }
+    }
+
+    /// Handle last version. Last version may be PUT or DELETE. If it is a PUT, value should be
+    /// load.
+    #[inline]
+    fn handle_last_version(
+        &mut self,
+        some_write: Option<Write>,
+        user_key: &Key,
+    ) -> Result<Option<Value>> {
+        match some_write {
+            None => Ok(None),
+            Some(write) => match write.write_type {
+                WriteType::Put => Ok(Some(self.reverse_load_data_by_write(write, user_key)?)),
+                WriteType::Delete => Ok(None),
+                _ => unreachable!(),
+            },
+        }
+    }
+
+    /// Load the value by the given `some_write`. If value is carried in `some_write`, it will be
+    /// returned directly. Otherwise there will be a default CF look up.
+    ///
+    /// The implementation is similar to `PointGetter::load_data_by_write`.
+    #[inline]
+    fn reverse_load_data_by_write(&mut self, write: Write, user_key: &Key) -> Result<Value> {
```

Comment:
The reverse is redundant, since it is a member function of BackwardScanner.

Reply:
I prefer to keep the name, otherwise some one may think that these "load_data_by_write" are the same (in fact they only differs in 5 bytes).